bindings/nodejs/src/client.rs

```diff
@@ -60,71 +58,69 @@ impl ClientMethodHandler {
-    crate::RUNTIME.spawn(async move {
-        *method_handler.write().await = None;
-        deferred.settle_with(&channel, move |mut cx| Ok(cx.undefined()));
-    });
-    Ok(promise)
+pub fn destroy_client(mut cx: FunctionContext) -> JsResult<JsUndefined> {
+    match cx.argument::<JsBox<SharedClientMethodHandler>>(0)?.write() {
+        Ok(mut lock) => *lock = None,
+        Err(e) => {
+            return cx
+                .throw_error(serde_json::to_string(&Response::Panic(e.to_string())).expect("json to string error"));
+        }
+    }
+    Ok(cx.undefined())
 }
 
 pub fn call_client_method(mut cx: FunctionContext) -> JsResult<JsUndefined> {
-    let method = cx.argument::<JsString>(0)?;
-    let method = method.value(&mut cx);
-    let method_handler = Arc::clone(&cx.argument::<JsBox<ClientMethodHandlerWrapper>>(1)?.0);
-    let callback = cx.argument::<JsFunction>(2)?.root(&mut cx);
-
-    let (sender, receiver) = std::sync::mpsc::channel();
-    crate::RUNTIME.spawn(async move {
-        if let Some(method_handler) = &*method_handler.read().await {
-            let (response, is_error) = method_handler.call_method(method).await;
-            method_handler.channel.send(move |mut cx| {
-                let cb = callback.into_inner(&mut cx);
-                let this = cx.undefined();
-
-                let args = [
-                    if is_error {
-                        cx.string(response.clone()).upcast::<JsValue>()
-                    } else {
-                        cx.undefined().upcast::<JsValue>()
-                    },
-                    cx.string(response).upcast::<JsValue>(),
-                ];
-
-                cb.call(&mut cx, this, args)?;
-
-                Ok(())
-            });
-        } else {
-            // Notify that the client got destroyed
-            // Safe to unwrap because the receiver is waiting on it
-            sender.send(()).unwrap();
+    match cx.argument::<JsBox<SharedClientMethodHandler>>(1)?.read() {
+        Ok(lock) => {
+            let method_handler = lock.clone();
+            let method = cx.argument::<JsString>(0)?;
+            let method = method.value(&mut cx);
+            let callback = cx.argument::<JsFunction>(2)?.root(&mut cx);
+            if let Some(method_handler) = method_handler {
+                crate::RUNTIME.spawn(async move {
+                    let (response, is_error) = method_handler.call_method(method).await;
+                    method_handler.channel.send(move |mut cx| {
+                        let cb = callback.into_inner(&mut cx);
+                        let this = cx.undefined();
+
+                        let args = [
+                            if is_error {
+                                cx.string(response.clone()).upcast::<JsValue>()
+                            } else {
+                                cx.undefined().upcast::<JsValue>()
+                            },
+                            cx.string(response).upcast::<JsValue>(),
```

Comment:
just for my understanding ... as it's wasn't changed in this PR ... but why do we need line 101?

Reply:
```
callClientMethod(method, handler, (error: Error, result: string)
```

The TS expects that tuple so this is building that.